src/nts_ke/server/connection.rs

```diff
@@ -95,6 +110,9 @@ pub struct KeServerConn {
     /// The status of the connection.
     state: KeServerConnState,
 
+    /// The state of NTS-KE.
+    ntske_state: RecievedNtsKeRecordState,
```

Comment:
Why don't you want to use `KeServerConnState`? It seems like we have two state fields now.

Reply:
Sorry, I gave a confusing name.
`RecievedNtsKeRecordState` is used for the store of received NTS-KE Records.
I think it not good that the status of the connection(=`KeServerConnState`) and the store of received NTS-KE Records(=`KeServerConnState`) be integrated.
It is needed rename. For example, `KeServerConnStore`.

Reply:
What do you think if `KeServerConnState` could look like this:
```
pub enum KeServerConnState {
    /// The connection is just connected. The TLS handshake is not done yet.
    Connected,
    /// Doing the TLS handshake,
    TlsHandshaking,
    /// The TLS handshake is done. It's opened for requests now.
    Opened,
    /// The request is received.
    RequestReceived(Vec<KeRecord>),
    /// The reponse is sent after getting a good request.
    ResponseSent,
    /// The connection is closed.
    Closed,
}

```

Reply:
It should be `RequestReceived(ClientRequest)` or something sorry.

Reply:
It is straightforward to record is store when `process_record` called.
Not used Vec, what do you think about the following?

```
pub enum KeServerConnState {
    /// The connection is just connected. The TLS handshake is not done yet.
    Connected,
    /// Doing the TLS handshake,
    TlsHandshaking,
    /// The TLS handshake is done. It's opened for requests now.
    Opened,
    /// The store of received NTS-KE Records.
    ReceivedNtsKeRecordState,
    /// The response is sent after getting a good request.
    ResponseSent,
    /// The connection is closed.
    Closed,
}
```

Reply:
Yes, but don't you want some data attached to that enum variant?

Reply:
I rethought about introducing enum such as `RequestReceived(Vec<KeRecord>)` or `RequestReceived(ReceivedNtsKeRecordState)`.

When introduced `RequestReceived(Vec<KeRecord>)` as an enum, It would need to extract `Vec<KeRecord>` to push records after call `deserialize()`.
Having two state fields seems to be better than storing received records as an enum, I think.

Reply:
Why do you think extracting is hard?

Reply:
Excuse me, there was not enough explanation. 
No, I think that having two state fields is better than extracting.
Because, If `nts_ke/server` had two state fields, it would not need to extract that once had stored to `RequestReceived`.

Reply:
Could you try to put in it the state please? I think renaming it to `ClientRequest` and put it as a variant `RequestReceived(ClientRequest)` of the enum `KeServerConnState` would be nice.

Please also keep in mind that it's better to avoid using `Option`. When you use it, please make sure that you really need to use it.

Reply:
I'm sorry for the delay.

Reply:
Thank you for comments! I will try it.